providers/dns/acmedns/internal/http_storage.go

```diff
@@ -17,6 +18,8 @@ import (
 
 var _ goacmedns.Storage = (*HTTPStorage)(nil)
 
+var ErrCNAMECreated = errors.New("the CNAME has already been created")
```

Comment:
The naming suggests that is an error *to create* a CNAME.

How about `ErrCNAMEAlreadyCreated` or `ErrDuplicateCNAME`?